lib/improver/blending/weights.py

```diff
@@ -834,15 +611,7 @@ def process(self, cube, coord_name, coord_vals=None, coord_unit='no_unit',
                    ' {0:s}'.format(str(type(cube))))
             raise TypeError(msg)
 
-        (num_of_weights,
-         exp_coord_found) = WeightsUtilities.process_coord(
-             cube, coord_name, coord_vals, coord_unit)
-
-        weights_in = self.nonlinear_weights(num_of_weights)
-
-        weights = WeightsUtilities.redistribute_weights(
-            weights_in, exp_coord_found, weights_distrib_method)
-
+        weights = self.nonlinear_weights(len(cube.coord(coord_name).points))
```

Comment:
Note to self: the checking for non-existent blending coordinates is performed MergeCubesForWeightedBlending and so is redundant here.